OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -78,10 +78,10 @@ protected virtual Activity InnerTick(Actor self, AttackBase attack)
 				return NextActivity;
 
 			var type = Target.Type;
-			if (!Target.IsValidFor(self) || type == TargetType.FrozenActor)
+			if (!Target.IsValidFor(self))
 				return NextActivity;
 
-			if (attack.Info.AttackRequiresEnteringCell && !positionable.CanEnterCell(Target.Actor.Location, null, false))
+			if (attack.Info.AttackRequiresEnteringCell && !positionable.CanEnterCell(Target.Location, null, false))
```

Comment:
A simpler fix to the sand worm issue will be to use `positionable.CanEnterCell(self.World.Map.CellContaining(Target.CenterPosition), null, false)`.